### PR TITLE
SFR-2304_Add Airtable API Authentication Flow

### PR DESCRIPTION
**Pull request by mitri-slory**
This PR adds a new authentication function to the process/util directory that is called in this early stage of the pubBacklist ingest process that will be developed in future tickets.

### CODE REVIEW

File: processes/ingest/pub_backlist.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import os
+import requests
+from ..util import airtable_authentication
+
+from ..core import CoreProcess
+from logger import createLog
+
+logger = createLog(__name__)
+
+class PubBacklistProcess(CoreProcess):
```
Comment (kylevillegas93) [inline processes/ingest/pub_backlist.py:10]: Let's be explicit: `PublisherBacklistProcess`

File: processes/ingest/pub_backlist.py
Comment (kylevillegas93) [inline processes/ingest/pub_backlist.py:1]: same with the file name - let's do publisher_backlists.py

File: processes/ingest/pub_backlist.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    def __init__(self, *args):
```
Comment (kylevillegas93) [inline processes/ingest/pub_backlist.py:11]: let's keep snake case for variables where we can.

File: processes/ingest/pub_backlist.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        self.createSession()
+
+        self.s3Bucket = os.environ['FILE_BUCKET']
+        self.createS3Client()
+
+    def runProcess(self):
+        try:
+            headers = airtable_authentication.create_authorization_header()
+
+            response = requests.get('https://api.airtable.com/v0/appBoLf4lMofecGPU/Publisher%20Backlists%20%26%20Collections%20%F0%9F%93%96?view=UofMichigan%20Press&maxRecords=3', headers=headers)
```
[the remark this answers is not in the document]
Comment (mitri-slory) [reply processes/ingest/pub_backlist.py:28]: I'll put the Airtable integration with the authentication in the util function I made. The service you are referring would be similar to the NYPLBibService right?

File: processes/ingest/pub_backlist.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+            print(response.json())
+
+        except Exception as e:
+            logger.exception('Failed to run Pub Backlist process')
+            raise e
+
```
[the remark this answers is not in the document]
Comment (mitri-slory) [reply processes/ingest/pub_backlist.py:35]: Line 35 is the new line. Do you want me to delete it?

File: processes/ingest/pub_backlist.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kylevillegas93) [reply processes/ingest/pub_backlist.py:28]: that's correct yes!

File: processes/ingest/pub_backlist.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kylevillegas93) [reply processes/ingest/pub_backlist.py:35]: ah hmm - maybe github is not recognizing that there is a new line! if there one at the of the file all good

File: processes/ingest/pub_backlist.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mitri-slory) [reply processes/ingest/pub_backlist.py:28]: Ok I'll most likely put the airtable integration and auth in a service together at a later point when the pub backlist ingestion process is more fleshed out.

**Review by kylevillegas93 — APPROVED** [review]
I think this is a good start! let's get it out the door and we can iterate on it from there!

**Pull request merged**